api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/AddressTest.java

```diff
@@ -0,0 +1,109 @@
+
+    @Test
+    void shouldValidateAddressSuccessfully() {
+        //Given
+        var address = aValidAddress();
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasNoViolations();
+    }
+
+    @Test
+    void shouldValidateAddressSuccessfullyWithNoAddressLine2() {
+        //Given
+        var address = anAddressWithAddressLine2(null);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasNoViolations();
+    }
+
+    @Test
+    void shouldFailToValidateAddressWithNoAddressLine1() {
+        //Given
+        var address = anAddressWithAddressLine1(null);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("must not be null", "addressLine1");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWhenAddressLine1IsTooLong() {
+        //Given
+        var address = anAddressWithAddressLine1(LONG_STRING);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("size must be between 1 and 500", "addressLine1");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWhenAddressLine2IsTooLong() {
+        //Given
+        var address = anAddressWithAddressLine2(LONG_STRING);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("size must be between 0 and 500", "addressLine2");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWithNoTownOrCity() {
+        //Given
+        var address = anAddressWithTownOrCity(null);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("must not be null", "townOrCity");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWhenTownOrCityIsTooLong() {
+        //Given
+        var address = anAddressWithTownOrCity(LONG_STRING);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("size must be between 1 and 500", "townOrCity");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWithNoPostcode() {
+        //Given
+        var address = anAddressWithPostcode(null);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("must not be null", "postcode");
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"AA1122BB", "A", "11AA21", "", "E!", "EA123"})
```

Comment:
Can we have a blank String in here too please?

Reply:
There is, fourth argument.